contracts/minter-suite/Minters/MinterHolder/MinterHolderV2.sol

```diff
@@ -0,0 +1,765 @@
+        );
+        return tokenId;
+    }
+
+    /**
+     * @notice gas-optimized version of purchase(uint256,address,uint256).
+     */
+    function purchase_nnf(
+        uint256 _projectId,
+        address _ownedNFTAddress,
+        uint256 _ownedNFTTokenId
+    ) external payable returns (uint256 tokenId) {
+        tokenId = purchaseTo_L69(
+            msg.sender,
+            _projectId,
+            _ownedNFTAddress,
+            _ownedNFTTokenId,
+            address(0)
+        );
+        return tokenId;
+    }
+
+    /**
+     * @notice Purchases a token from project `_projectId` and sets
+     * the token's owner to `_to`.
+     * @param _to Address to be the new token's owner.
+     * @param _projectId Project ID to mint a token on.
+     * @param _ownedNFTAddress ERC-721 NFT address owned by msg.sender being used to
+     * claim right to purchase.
+     * @param _ownedNFTTokenId ERC-721 NFT token ID owned by msg.sender being used
+     * to claim right to purchase.
+     * @return tokenId Token ID of minted token
+     */
+    function purchaseTo(
+        address _to,
+        uint256 _projectId,
+        address _ownedNFTAddress,
+        uint256 _ownedNFTTokenId
+    ) external payable returns (uint256 tokenId) {
+        return
+            purchaseTo_L69(
+                _to,
+                _projectId,
+                _ownedNFTAddress,
+                _ownedNFTTokenId,
+                address(0)
+            );
+    }
+
+    /**
+     * @notice Purchases a token from project `_projectId` and sets
+     *         the token's owner to `_to`, as a delegate, (the `msg.sender`)
+     *         on behalf of an explicitly defined vault.
+     * @param _to Address to be the new token's owner.
+     * @param _projectId Project ID to mint a token on.
+     * @param _ownedNFTAddress ERC-721 NFT address owned by msg.sender being used to
+     * claim right to purchase.
+     * @param _ownedNFTTokenId ERC-721 NFT token ID owned by msg.sender being used
+     * to claim right to purchase.
+     * @param _vault Vault being purchased on behalf of.
+     * @return tokenId Token ID of minted token
+     */
+    function purchaseTo(
+        address _to,
+        uint256 _projectId,
+        address _ownedNFTAddress,
+        uint256 _ownedNFTTokenId,
+        address _vault
+    ) external payable returns (uint256 tokenId) {
+        return
+            purchaseTo_L69(
+                _to,
+                _projectId,
+                _ownedNFTAddress,
+                _ownedNFTTokenId,
+                _vault
+            );
+    }
+
+    /**
+     * @notice gas-optimized version of purchaseTo(address,uint256,address,uint256,address).
+     */
+    function purchaseTo_L69(
+        address _to,
+        uint256 _projectId,
+        address _ownedNFTAddress,
+        uint256 _ownedNFTTokenId,
+        address _vault // acceptable to be `address(0)` if no vault
+    ) public payable nonReentrant returns (uint256 tokenId) {
```

Comment:
From checking against https://emn178.github.io/solidity-optimize-name/ I think we want to update our gas-optimized name from the previous `purchaseTo_L69` to be now `purchaseTo_dlc`^.

^You can verify this yourself by passing `purchaseTo(address,uint256,address,uint256,address)` into that tool–tool is a bit poorly documented in terms of what its expected input format is.

Reply:
++ to the above, and would like to also have natspec comments on this public function (can re-use one from `purchaseTo`)